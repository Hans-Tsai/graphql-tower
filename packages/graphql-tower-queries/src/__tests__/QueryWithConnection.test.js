import _ from 'lodash';
import faker from 'faker';
import { graphql, GraphQLInt, GraphQLSchema, GraphQLObjectType } from 'graphql';
import { QueryWithConnection } from '../';

describe('QueryWithConnection', () => {
  it('QueryWithConnection', async () => {
    const resolve = jest.fn();
    const QueryConnection = class extends QueryWithConnection {
      type = GraphQLInt;
      args = { id: { type: GraphQLInt } };
      resolve = resolve;
    };

    const query = new QueryConnection();
    expect(query).toMatchSnapshot();

    expect(query.node).toEqual(GraphQLInt);

    const queryExtend = _.extend({}, query);
    expect(queryExtend.node).toBeUndefined();
    expect(queryExtend.args).not.toBeUndefined();
    expect(queryExtend.type).not.toBeUndefined();
    expect(queryExtend.resolve).not.toBeUndefined();

    const first = faker.random.number();
    const offset = faker.random.number();
    const after = faker.lorem.word();
    const schema = new GraphQLSchema({
      query: new GraphQLObjectType({
        name: 'Query',
        fields: { connection: query },
      }),
    });

    const reply = [faker.random.number(), faker.random.number()];

    resolve.mockReturnValueOnce(reply);
    const result = await graphql(schema, `
      query ($first: Int $offset: Int $after: String) {
        connection (first: $first offset: $offset after: $after)
      }`, {}, {}, { first, offset, after });

    expect(result.data.connection).toEqual(reply);
  });
});
