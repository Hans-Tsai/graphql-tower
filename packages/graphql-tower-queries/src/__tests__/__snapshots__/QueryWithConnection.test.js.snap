// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QueryWithConnection QueryWithConnection 1`] = `
QueryConnection {
  "_": Object {
    "afterware": Array [],
    "args": Object {
      "after": Object {
        "type": "String",
      },
      "first": Object {
        "type": "Int",
      },
      "id": Object {
        "type": "Int",
      },
      "offset": Object {
        "type": "Int",
      },
    },
    "middleware": Array [],
    "node": "Int",
    "parentArgs": undefined,
    "resolve": [Function],
    "type": "[Int]",
  },
  "args": Object {
    "after": Object {
      "type": "String",
    },
    "first": Object {
      "type": "Int",
    },
    "id": Object {
      "type": "Int",
    },
    "offset": Object {
      "type": "Int",
    },
  },
  "resolve": [Function],
  "type": "[Int]",
}
`;
